lib/cli-engine.js

```diff
@@ -496,15 +517,19 @@ CLIEngine.prototype = {
             results.push(res);
         }
 
-        files.forEach(function(pattern) {
-            glob.sync(pattern, globOptions).forEach(executeOnFile);
+        patterns.forEach(function(pattern) {
+            if (shell.test("-f", pattern) && !ignoredPaths.contains(pattern)) {
```

Comment:
shelljs is only used in here at 2 spots where we can easily use `fs`. By doing that we can move `shelljs` back to `devDependencies` sections